src/routes/shadowBanUser.js

```diff
@@ -47,15 +48,22 @@ module.exports = async function shadowBanUser(req, res) {
 
         //find all previous submissions and hide them
         if (unHideOldSubmissions) {
-        db.prepare('run', "UPDATE sponsorTimes SET shadowHidden = 1 WHERE userID = ?", [userID]);
+            db.prepare('run', "UPDATE sponsorTimes SET shadowHidden = 1 WHERE userID = ?", [userID]);
         }
     } else if (!enabled && row.userCount > 0) {
         //remove them from the shadow ban list
         privateDB.prepare('run', "DELETE FROM shadowBannedUsers WHERE userID = ?", [userID]);
 
         //find all previous submissions and unhide them
         if (unHideOldSubmissions) {
-            db.prepare('run', "UPDATE sponsorTimes SET shadowHidden = 0 WHERE userID = ?", [userID]);
+            let segmentsToIgnore = db.prepare('all', "SELECT uuid FROM sponsorTimes st JOIN noSegments ns on st.videoID = ns.videoID AND st.category = ns.category WHERE st.userID = ?", [userID]).map((item) => item.UUID);
+            let allSegments = db.prepare('all', "SELECT uuid FROM sponsorTimes st WHERE st.userID = ?", [userID]).map((item) => item.UUID);
+            
+            allSegments.filter((item) => {
+                return segmentsToIgnore.indexOf(item) === -1;
+            }).forEach((uuid) => {
+                db.prepare('run', "UPDATE sponsorTimes SET shadowHidden = 0 WHERE uuid = ?", [uuid]);
+            });
```

Comment:
I feel like this might be more efficient if it is al in one query,, but it should be fine since this is only run by admins.